futures-util/src/future/try_future/mod.rs

```diff
@@ -449,7 +526,7 @@ pub trait TryFutureExt: TryFuture {
     /// assert_eq!(list, Ok(vec![17, 18, 19]));
     /// # });
     /// ```
-    fn try_flatten_stream(self) -> TryFlattenStream<Self>
+    fn try_flatten_stream(self) -> TryFlattenStream<Self, Self::Ok>
```

Comment:
Do you think we can avoid this change?

Reply:
Yep 👍 

One thing I found was that the code is quite inconsistent in when it uses type parameters vs extra bounds. (Ie. in this case, removing the extra type parameter means putting an extra bound on the `Fut` type).

It seems in some places it tries to avoid bounds on the types themselves (which does make them easier to use as you don't have to repeat those bounds everywhere) but in other places it prefers to reduce the number of type parameters.

Luckily, with the macro system it is very easy to switch back and forth.